### PR TITLE
fix: cutomer duplication issue fixed

### PR DESCRIPTION
**Pull request by vishdha**
TASK ID: https://bloomstack.com/desk#Form/Issue/ISS-2022-00001

issue: on_session_start it execute this command erpnext.selling.utils.initialise_order_for. when we delete a customer from the contact links table and if there is email_id present in the email child table. it tries to create a new customer and contact for that customer using the same customer parameter.
That is resolved

before:
Peek 2022-01-12 14-08

after:
Peek 2022-01-12 14-06

### CODE REVIEW

File: erpnext/portal/utils.py [hunk 1]
```diff
@@ -112,11 +112,18 @@ def create_customer_or_supplier():
...
 		contact = frappe.get_doc('Contact', contact_name)
 		doctypes = [d.link_doctype for d in contact.links]
-		return doctype in doctypes
-
-	return False
+		if doctypes:
+			return doctype in doctypes
+		else:
+			fullname = frappe.utils.get_fullname(user)
+			return frappe.db.exists(doctype, fullname)
```
Comment (sahil28297) [inline erpnext/portal/utils.py:129]:
```suggestion
			return frappe.db.exists(doctype, fullname)
	return False
```

File: erpnext/selling/utils/order_for.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def initialise_order_for():
 			order_for["customer_primary_contact_name"] = contact.name
 
 	# Keep customer info up to date on every session start
-	customer = get_party()
+	customer = get_party(party_doctype="Customer", party=order_for.get("customer_name"))
```
Comment (sahil28297) [inline erpnext/selling/utils/order_for.py:42]: why this change?

**Review by sahil28297 — CHANGES_REQUESTED** [review]

File: erpnext/portal/utils.py (hunk 1 above)
Comment (vishdha) [reply erpnext/portal/utils.py:129]: Sure. got the point

File: erpnext/selling/utils/order_for.py (hunk 2 above)
Comment (vishdha) [reply erpnext/selling/utils/order_for.py:42]: It tries to create a new customer once if you look at cart.py. you will understand this

Comment (DeveloperMetal) [reply erpnext/selling/utils/order_for.py:42]: This makes sense, it enforces a customer type when using order-for functionality.

**Review by DeveloperMetal — DISMISSED** [review]

File: erpnext/portal/utils.py [hunk 3]
```diff
@@ -112,11 +112,20 @@ def create_customer_or_supplier():
 
 
 def party_exists(doctype, user):
+	""" To check if particular doctype eixsts or not. it will try to find links reference.
+	if link referene available then it will return true else
```
Comment (sahil28297) [inline erpnext/portal/utils.py]:
```suggestion
	if link reference available then it will return true else
```

**Review by sahil28297 — CHANGES_REQUESTED** [review]

**Review by sahil28297 — APPROVED** [review]

File: erpnext/portal/utils.py (hunk 3 above)
Comment (vishdha) [reply erpnext/portal/utils.py]: !
